File: Sliding Window/Longest Subarray of 1's After Deleting One Element/solution.go
package main

func longestSubarray(nums []int) int {
	max := 0
	start := 0
	idxZero := -1
	for i := 0; i < len(nums); i++ {
		if nums[i] == 0 {
			start = idxZero + 1
			idxZero = i
		}
		if i-start+1 > max {
			max = i - start + 1
		}
	}

	return max - 1
}

File: Sliding Window/Longest Subarray of 1's After Deleting One Element/solution.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        result = 0
        start = 0
        k = 0
        for i, num in enumerate(nums):
            if num == 0:
                k += 1
            while k > 1:
                if nums[start] == 0:
                    k -= 1
                start += 1
            result = max(result, i - start)
        return result
